### ISSUE TITLE
TraCI induction loop considers vehicles of the current time step

### ISSUE THREAD

**Issue by behrisch** [DLR-TS/sumo#3552]
it should only consider values of the last time step as stated in documentation and function names

**Comment by aminissn** [comment]
Hi Michael,
do you have any idea when this issue will be solved? I guess it is still unsolved for version 32!

Thanks!
Sasan

**Comment by behrisch** [comment]
No sorry, but I put it on my list.

**Comment by aminissn** [comment]
Is there a workaround for this bug? One other thing that can be nice to have is to get occupancy=100% when a non-moving vehicle stands on the detector. Currently, I get -1. I guess it gives values when getTimeSinceDetection is 0, which is not the case when there is congestion on the detector.

**Comment by behrisch** [comment]
@aminissn Could you test this or even better send a small test scenario?

**Comment by aminissn** [comment]
Here is the example I was testing with. I am basically comparing the FD from TraCI and loop detector output file. 
[01_onelane.zip](https://github.com/eclipse/sumo/files/3845413/01_onelane.zip)

**Comment by m-kro** [comment]
@behrisch: Due to your changes at [https://github.com/eclipse/sumo/blob/c31e7cbd20acffe7a4b9a0e5365190993351b2b3/src/microsim/output/MSInductLoop.cpp#L295](url), standing vehicles are only considered in one time step for e.g. TraCI inductionloop.getLastStepVehicleNumber. Only when it just has entered the detector, its enter time fulfills the condition. Is that really the desired behaviour?

**Comment by behrisch** [comment]
@m-kro No, it is not. It is very high on my ToDo-List for the new year to solve this ticket. My general approach would be to let the behavior mimic a real loop as close as possible, meaning all values which require that the vehicle has already left the loop (speed, length, ...) are only available when the vehicle has left. Everything else should move to the instant loop. @aminissn Do you agree here?

**Comment by aminissn** [comment]
@behrisch I agree on making the values available after the vehicle has left the detector.

**Comment by aminissn** [comment]
I totally forgot about this matter and was trying to implement an ALINEA controller and noticed the difference between the Fundamental Diagram obtained from TraCI and the generated E1 detector output. Just curios about the best way to get the "right" occupancy and flow measurements using TraCI for actuated control applications?
![image](https://user-images.githubusercontent.com/30176885/153222273-49bfd236-d0b1-42c4-a144-7508db9687e4.png)
